### ISSUE TITLE
Condition if stream is on or off

### ISSUE THREAD

**Issue by Trukess** [justintv/Twitch-API#675]
I was reading the API documentation from twitch but didn't find what I needed, I just need a condition that gives me true or false if streamer is on or off, someone can help me?

**Comment by freaktechnik** [comment]
You'll need https://dev.twitch.tv/docs/v5/reference/streams/.

Further this belongs to https://discuss.dev.twitch.tv and not an inactive bug tracker for obsolete API documentation.

**Comment by Trukess** [comment]
Thank you very much for answer me freaktechnik, where I should type:

curl -H 'Accept: application/vnd.twitchtv.v5+json' \
-H 'Client-ID: uo6dggojyb8d6soh92zknwmi5ej1q2' \
-X GET 'https://api.twitch.tv/kraken/streams/?game=Overwatch'

What kind of language is this?

**Comment by scagood** [comment]
curl is a command line tool. (https://curl.haxx.se/)

This means you'd put the following into your console:
![f41ee468e886559c0a197610d5afb0cc](https://cloud.githubusercontent.com/assets/2230835/24069678/74f2838a-0ba5-11e7-8857-ed0854bd57fe.png)

The top part is the command.
curl -H "Accept: application/vnd.twitchtv.v5+json" -H "Client-ID: uo6dggojyb8d6soh92zknwmi5ej1q2" -X GET "https://api.twitch.tv/kraken/streams/?game=Overwatch"
The next part is the response. 

Broken down this gives you:
The application: curl
Add a header: -H "Accept: application/vnd.twitchtv.v5+json"
Add a header -H "Client-ID: uo6dggojyb8d6soh92zknwmi5ej1q2"
Set the request method: -X GET
Finally the url you're requesting: 'https://api.twitch.tv/kraken/streams/?game=Overwatch'

Hope this helps,
sca

**Comment by Trukess** [comment]
Thanks for the answer scagood, so I need to install latest version from curl in my computer?

**Comment by freaktechnik** [comment]
No, that's just how the API documentation shows examples for making requests to the API. You can use whatever you want to make requests.

**Comment by Trukess** [comment]
Thanks for answer freaktechnik, I have the following function in php:

`function file_get_contents_curl($url) {
    $clientID = array(
        "Client-ID: xxxxxxx"
    );

    $ch = curl_init();

    curl_setopt($ch, CURLOPT_AUTOREFERER, TRUE);
    curl_setopt($ch, CURLOPT_HEADER, 0);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
    curl_setopt($ch, CURLOPT_URL, $url);
    curl_setopt($ch, CURLOPT_FOLLOWLOCATION, TRUE);
    curl_setopt($ch, CURLOPT_HTTPHEADER, $clientID);

    $data = curl_exec($ch);
    curl_close($ch);

    return $data;
}`

but didn't work, need I some plugin from curl?

**Comment by Trukess** [comment]
Oh I understand, thanks freaktechnik 🍡

**Comment by EpsilonAlpha** [comment]
@freaktechnik to determine if a channel is online i proof that 

".stream.viewers" is not NULL

 and in my case if

 ".stream.stream_type" is "live" 

so i skip out the Vodcasts.

Hopes that helps you.

**Update:** 
Currently the v5 API returns if a stream is offline:
{
  "stream": null
}

v3 version if a stream is offline
{
  "stream": null,
  "_links": {
    "self": "https://api.twitch.tv/kraken/streams/<streamernick>",
    "channel": "https://api.twitch.tv/kraken/channels/<streamernick>"
  }
}
